cpp/src/arrow/io/CMakeLists.txt

```diff
@@ -37,6 +38,7 @@ set(ARROW_IO_TEST_LINK_LIBS
   ${ARROW_IO_PRIVATE_LINK_LIBS})
```

Comment:
We would need a `arrow_io_static` here to solve the linking problems in the tests.